rfcs/test_harness/in_engine_test_harness.md

```diff
@@ -0,0 +1,402 @@
+  - Ignoring a preferred language will reduce the potential user base interested in using this feature
+    - C++ tests can be relatively performant, and easier to attach to lower-level inspection
+    - Lua and ScriptCanavs tests do not require recompilation, and can be more accessible to many developers
+- Only implement commandline test execcution, no Editor UI features
+  - Significantly reduces iteration speed for creating level-dependent tests
+  - Reduces discoverability, reducing the number of users who will try to run and author tests
+- Do not implement this, and leave it to contributors and customers to build what they want
+  - Writing in-game automation will continue to have a high startup cost
+    - O3DE more likely to follow its current inertia of few automated in-game tests
+    - Teams using O3DE will be less likely to invest in their own automated in-game tests
+
+## How will users learn this feature?
+
+Users will primarily encounter this feature in 4 locations:
+
+1. Overview and Getting Started pages in the O3DE.org documentation
+2. Example tests in O3DE's AutomatedTesting project, with inline "how and why" documentation
+3. Elements in the Editor UI (after enabling the Gem)
+4. The CLI tool to launch tests, which includes help information
+
+## Are there any open questions?
+
+### Interactive Socket Interface
+
+Should an interactive socket be used, or are CLI launch arguments sufficient?
+
+Using sockets would simplify writing networked test scenarios which sequentially step through commands between multiple devices. In the diagram below sockets could follow either orchestration-script writing pattern, whereas CLI parameters could only use the pattern on the right:
+
+![Networked scripts](netscript.png?raw=true)
+
+If a socket connection is not used, it will be more difficult to coordinate tests and report results. Not using a socket will also require any test filtering and iteration occur inside the Engine, increasing the complexity of what must run in each application.
+
+The need to frequently relaunch the application with new CLI Args could be mitigated by having the EditorSystemComponent save a manifest for which tests exist. Without the manifest, even just listing out which tests are available would require the application be started and a specific level loaded (potentially every level).
+
+#### Interactive Socket Pros
+
+- Reduces need to re-launch the application to run multiple tests
+  - CLI arg buffer has size limits dependent on OS/target settings, which can impact large command strings
+- Sending test status messages over a socket reduces reliance on inspecting log files
+  - Mobile device logs can be difficult to interact with
+- Can provide a secondary health check to help detect if the application has become nonresponsive
+- Easier to write scripts for networked tests
+
+#### Interactive Socket Cons
+
+- Slightly more complex to implement and maintain
+- Larger security footprint
+
+#### Remote Console
```

Comment:
Another alternative worth investigating is the Remote Tools Gem: https://github.com/o3de/o3de/tree/development/Gems/RemoteTools